cutadapt/modifiers.py

```diff
@@ -273,3 +281,59 @@ def __call__(self, read):
 		start_cut = start_cut.end() if start_cut else 0
 		end_cut = end_cut.start() if end_cut else len(read)
 		return read[start_cut:end_cut]
+
+
+class Modifiers(object):
+	def __init__(self):
+		self.mod1 = OrderedDict()
+		self.mod2 = OrderedDict()
+	
+	def add_modifier(self, mod_type, read=1, **kwargs):
+		mod = ModType.create_modifier(mod_type, **kwargs)
+		if read & 1 > 0:
+			self.mod1[mod_type] = mod
+		if read & 2 > 0:
+			self.mod2[mod_type] = mod
+	
+	def modify(self, read1, read2=None):
+		for mod in self.mod1.values():
+			read1 = mod(read1)
+		if read2:
+			for mod in self.mod2.values():
+				read2 = mod(read2)
+			return (read1, read2)
+		else:
+			return read1
+
+
+class _ModType(dict):
+	"""
+	Enumeration of modifier types.
+	
+	This is an enum-like class that is compatible with python 2x. 
+	This class is only meant to be instantiated once, below (this 
+	would be an anonymous class, if python allowed such a thing).
```

Comment:
The sentence "this would be an anonymous class" shouldn't be there. That’s for a Git commit message or a blog entry.